Content.Server/GameObjects/Components/Markers/ConditionalSpawnerComponent.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using Content.Server.GameTicking;
+using Content.Server.Interfaces.GameTicking;
+using NFluidsynth;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.Interfaces.Reflection;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using SQLitePCL;
+using Logger = Robust.Shared.Log.Logger;
+
+namespace Content.Server.GameObjects.Components.Markers
+{
+    [RegisterComponent]
+    public class ConditionalSpawnerComponent : Component, IMapInit
+    {
+        public override string Name => "ConditionalSpawner";
+
+#pragma warning disable 649
+        [Dependency] private IGameTicker _gameTicker;
+        [Dependency] private IReflectionManager _reflectionManager;
+        [Dependency] private IEntityManager _entityManager;
+        [Dependency] private IRobustRandom _robustRandom;
+#pragma warning restore 649
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public List<string> Prototypes { get; set; }
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        private List<string> _gameRules = new List<string>();
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float Chance { get; set; } = 1.0f;
+
+        public List<Type> GameRules
+        {
+            get
+            {
+                var list = new List<Type>();
+
+                if(_gameRules != null)
+                    foreach (var rule in _gameRules)
+                    {
+                        list.Add(_reflectionManager.GetType(rule));
+                    }
+
+                return list;
+            }
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(this, x => Prototypes, "prototypes", null);
+            serializer.DataField(this, x => Chance, "chance", 1.0f);
+            serializer.DataField(this, x => _gameRules, "gameRules", null);
+        }
+
+        private void RuleAdded(GameRuleAddedEventArgs obj)
+        {
+            if(_gameRules == null || _gameRules.Count > 0)
+                TrySpawn();
+        }
+
+        private void TrySpawn()
+        {
+            if (_gameRules == null || _gameRules.Count == 0)
+            {
+                // ReSharper disable twice CompareOfFloatsByEqualityOperator
+                if(Chance == 1.0f)
```

Comment:
? If the chance of spawning is 1, then do nothing?

Reply:
Aah, that was actually a leftover from when this component could only spawn single prototypes instead of choosing from a list. Removed now!